File: Tests/Reqnroll.Formatters.Tests/Samples/Resources/global-hooks-afterall-error/global_hooks_afterall_error.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Reqnroll;

namespace Reqnroll.Formatters.Tests.Samples.Resources.global_hooks_afterall_error;

[Binding]
public class global_hooks_afterall_error
{
    [BeforeTestRun]
    public static void BeforeTestRun1()
    {

    }

    [BeforeTestRun]
    public static void BeforeTestRun2()
    {

    }

    [When("a step passes")]
    public void AStepPasses()
    {

    }

    [AfterTestRun]
    public static void AfterTestRun1()
    {

    }

    [AfterTestRun]
    public static void AfterTestRun2()
    {
        throw new ApplicationException("AfterAll hook went wrong");
    }

    [AfterTestRun]
    public static void AfterTestRun3()
    {

    }
}

File: Tests/Reqnroll.Formatters.Tests/Samples/Resources/global-hooks-afterall-error/global_hooks_afterall_error.ts
import { When, BeforeAll, AfterAll } from '@cucumber/fake-cucumber'

BeforeAll({}, function () {
  // no-op
})

BeforeAll({}, function () {
  // no-op
})

When('a step passes', function () {
  // no-op
})

AfterAll({}, function () {
  // no-op
})

AfterAll({}, function () {
  throw new Error('AfterAll hook went wrong')
})

AfterAll({}, function () {
  // no-op
})
